doc/wg/network/notes/network-notes-2023-08-10.md

```diff
@@ -0,0 +1,96 @@
+ * Felix: Most of the stack is in software, but I have two hardware abstraction layers. One for the universal way of working with CAN hardware, one for very specific hardware where the manufacturer changed a lot of things. That latter HAL just has a simple "send" option.
+ * Branden: What does your userspace interface look like?
+ * Felix: I have a couple of buffers where userspace can put frames. Interface just sends. On receiving, most frames just arrive into buffers.
+ * Branden: What does the interface to userspace look like?
+ * Felix: There is a special interface that userspace adds which has a header that the capsule decodes. The capsule then encapsulates the data from the userspace and figures out how to send it.
+ * Leon: Do you have any thoughts on interoperability between Ethernet and CAN?
+ * Felix: CAN is really just a way to send 64 bits. You have to decide what the bits mean. There is a CAN transfer protocol for sending huge frames. It still doesn't define meaning of bits though. So for Ethernet to CAN, you just stuff the ethernet packet bits in CAN.
+ * Leon: So for bridging, I would make my own encapsulation protocol with headers, and then make my own translation component which translates the frames.
+ * Felix: Yes, sort of. In AUTOSAR there is some logic. Certain signals correspond to certain collections of bits.
+ * Leon: So a question from Tockworld I had was how to make an interface to bridge protocols. So I'm interested here
+ * Alex: I think in cars they use mostly UDP. So they take the frame from CAN and drop it in a UDP frame. Then they just need to add retransmission, for some types of packets, others they don't care if they lose.
+ * Branden: Do you have support for large packet types?
+ * Felix: Not at this time, although we can fragment across small packets
+
+## LoRa
+ * Branden: The existing LoRa support in Tock is entirely Alistair's effort, but I'll say some things on it. LoRa is long-range wireless (kilometer range) while still at low energy and at low bitrate (kbps). There only exists really one family of chips for LoRa communication, which are usually communicated with over SPI. So Alistair's efforts connect a raw SPI interface to userland, and there are C libraries to drive the LoRa radios over SPI. This would usually have problems with timing, but LoRa is very slow, with a full second between reception and an ACK, for instance. So this works.
+ * Alex: A LoRa gateway could be an interesting connection of networks here: LoRa + Ethernet
```

Comment:
I wanted to jump in and say I don't think Tock is a good fit for a LoRa gateway. Gateways tend to be pretty big systems, it really is job for Linux on a RaspberryPi at a minimum.

Tock for P2P LoRa and even LoRaWAN devices I think is a really cool usecase. There are pretty large projects like [Meshtastic](https://meshtastic.org/) using LoRa P2P, so it's very handy.

RadioLib (which is the userspace application) [supports LoRaWAN](https://github.com/jgromes/RadioLib/blob/master/examples/LoRaWAN/LoRaWAN_End_Device/LoRaWAN_End_Device.ino). So I'm hoping to support Tock devices as LoRaWAN clients. Then we can have smart Tock sensors transmitting data to TTN!

I already have [an example](https://www.alistair23.me/2023/08/09/lora-on-sparkfun-board/) of transmitting sensor data between Tock instances up and running

Reply:
Thanks for sharing this. I use RadioLib for my classes and it's a really high-quality project. I didn't know they had added LoRaWAN support. And the Tock support you added to it is fantastic!!